### PR TITLE
Blit directly to texture

### PR DESCRIPTION
**Pull request by fabiangreffrath**
https://github.com/fabiangreffrath/woof/pull/1870

### CODE REVIEW

**Comment by fabiangreffrath** [comment]
@gendlin @rfomin @ceski-1 
Which other optimizations to Woof can you think of that we could contribute back to Choco?

**Comment by gendlin** [comment]
fabiangreffrath/woof#1870
fabiangreffrath/woof#1761 (minor)

**Comment by fabiangreffrath** [comment]
> [fabiangreffrath/woof#1810](https://github.com/fabiangreffrath/woof/pull/1810)

https://github.com/chocolate-doom/chocolate-doom/pull/1701

File: src/i_video.c [hunk 1]
```diff
@@ -1358,12 +1356,8 @@ static void SetVideoMode(void)
...
     if (argbbuffer == NULL)
     {
-        SDL_PixelFormatEnumToMasks(pixel_format, &bpp,
-                                   &rmask, &gmask, &bmask, &amask);
-        argbbuffer = SDL_CreateRGBSurface(0,
-                                          SCREENWIDTH, SCREENHEIGHT, bpp,
-                                          rmask, gmask, bmask, amask);
-        SDL_FillRect(argbbuffer, NULL, 0);
+        argbbuffer = SDL_CreateRGBSurfaceWithFormatFrom(
+                     NULL, w, h, 8, 0, SDL_PIXELFORMAT_ARGB8888);
```
Comment (gendlin) [inline src/i_video.c:1360]: The `8` here from the original code was not really correct (my bad). It can be safely replaced with `0` (SDL ignores it)

**Comment by turol** [comment]
This does not seem to match SDL2 documentation. 

https://wiki.libsdl.org/SDL2/SDL_CreateRGBSurfaceWithFormatFrom seems to indicate that the first parameter should be a pointer to existing buffer.

**Comment by ceski-1** [comment]
> Which other optimizations to Woof can you think of that we could contribute back to Choco?

Both Crispy and Woof have the option to disable smooth pixel scaling which is a nice performance boost for older computers at the expense of uneven pixels. There's a feature request for this as well: https://github.com/chocolate-doom/chocolate-doom/issues/1686

**Comment by fabiangreffrath** [comment]
> This does not seem to match SDL2 documentation.
> 
> https://wiki.libsdl.org/SDL2/SDL_CreateRGBSurfaceWithFormatFrom seems to indicate that the first parameter should be a pointer to existing buffer.

That's an SDL documentation issue then. The function just passes the pointer to the created surface without any problems:

https://github.com/libsdl-org/SDL/blob/ea820edc62fd1148ebea35c000c796b689bcb2e4/src/video/SDL_surface.c#L257

**Comment by gendlin** [comment]
> Both Crispy and Woof have the option to disable smooth pixel scaling which is a nice performance boost for older computers at the expense of uneven pixels. There's a feature request for this as well: #1686

If it was just hard-switched to nearest filtering I wonder if anyone would notice or complain? The extra codepaths required for what's just a slower, blurrier mode don't seem worth the benefit to me...but I admittedly hate linear filtering, so :)

**Comment by fabiangreffrath** [comment]
I have added a comment to make this clear.

**Pull request merged**